### PR TITLE
Adding Task server backend changes for Hue

### PR DESCRIPTION
**Pull request by ranade1**
This change enables Hue backend server, following hue.ini parameters are needed.

[desktop]
enable_chunked_file_uploader=true
[filebrowser]
max_file_size_upload_limit=-1 # Bytes
[[task_server]]
enabled=True
beat_enabled=True
fetch_result_limit=20000
result_storage='{"BACKEND": "django.core.files.storage.FileSystemStorage", "properties": {"location": "./logs"}}' broker_url="redis://localhost:6379/0"
celery_result_backend="redis://localhost:6379/0"
celeryd_opts='--time-limit=300'
execution_storage='{"BACKEND": "django.core.cache.backends.locmem.LocMemCache", "LOCATION": "celery-hue"}'

Testing Done:
Tested file upload on HDFS, S3, ABFS, Ozone system

## What changes were proposed in this pull request?

- (Please fill in changes proposed in this fix)

## How was this patch tested?

- (Please explain how this patch was tested. Ex: unit tests, manual tests)
- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Review by wing2fly — APPROVED** [review]

File: apps/filebrowser/src/filebrowser/views.py [hunk 1]
```diff
@@ -1479,12 +1479,29 @@ def perform_upload(request, *args, **kwargs):
   """
   scheme = get_scheme(kwargs['dest'])
   upload_class = UPLOAD_CLASSES.get(scheme, LocalFineUploaderChunkedUpload)
-  _fs = upload_class(request, **kwargs)
-  _fs.upload()
-  if scheme == 'hdfs':
-    result = _massage_stats(request, stat_absolute_path(_fs.filepath, request.fs.stats(_fs.filepath)))
+  result = None
+
+  if TASK_SERVER.ENABLED.get():
```
Comment (JohanAhlen) [inline apps/filebrowser/src/filebrowser/views.py:1484]: How about a unit test for this?

**Review by athithyaaselvam — APPROVED** [review]

File: apps/filebrowser/src/filebrowser/views.py [hunk 2]
```diff
@@ -1479,12 +1479,29 @@ def perform_upload(request, *args, **kwargs):
...
+    # If task server is enabled, upload the file to the task server.
+    print("Uploading file to task server")
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/views.py:1486]: Remove print statement?

File: apps/filebrowser/src/filebrowser/views.py [hunk 3]
```diff
@@ -1479,12 +1479,29 @@ def perform_upload(request, *args, **kwargs):
...
+    _fs = upload_class(request, **kwargs)
+    _fs.check_access()
+    from filebrowser.tasks import upload_file_task, error_handler
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/views.py:1489]: Imports at the top or its a circular dependency?

File: apps/filebrowser/src/filebrowser/views.py (hunk 1 above)
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/views.py:1484]: +1

File: apps/filebrowser/src/filebrowser/views.py [hunk 4]
```diff
@@ -1479,12 +1479,29 @@ def perform_upload(request, *args, **kwargs):
...
+    result = "task started %s" % task_id
+    logger.info("Task started %s" % task_id)
   else:
-    result = "success"
+    _fs = upload_class(request, **kwargs)
+    _fs.upload()
+    if scheme == 'hdfs':
+      result = _massage_stats(request, stat_absolute_path(_fs.filepath, request.fs.stats(_fs.filepath)))
+    else:
+      result = "success"
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/views.py:1504]: Curious: Why we are doing this for hdfs?

File: desktop/core/src/desktop/lib/fs/ozone/upload.py [hunk 5]
```diff
@@ -45,6 +46,8 @@ def __init__(self, request, *args, **kwargs):
     if self.file_name:
       self.file_name = unicodedata.normalize('NFC', self.file_name) # Normalize unicode
     self.chunk_size = UPLOAD_CHUNK_SIZE.get()
+    if kwargs.get('chunk_size', None) != None:
```
Comment (Harshg999) [inline desktop/core/src/desktop/lib/fs/ozone/upload.py:49]: Using `if kwargs.get('chunk_size'):` should simplify the check?

File: desktop/core/src/desktop/settings.py [hunk 6]
```diff
@@ -747,8 +747,27 @@ def disable_database_logging():
   CELERY_ACCEPT_CONTENT = ['json']
   CELERY_RESULT_BACKEND = desktop.conf.TASK_SERVER.CELERY_RESULT_BACKEND.get()
   CELERY_TASK_SERIALIZER = 'json'
+  CELERY_ENABLE_UTC = True
+  CELERY_TIMEZONE = "America/Los_Angeles"
```
Comment (Harshg999) [inline desktop/core/src/desktop/settings.py:751]: Curious: Why not take it from Hue config?

File: desktop/libs/aws/src/aws/s3/upload.py [hunk 7]
```diff
@@ -64,13 +64,14 @@ def __init__(self, request, *args, **kwargs):
...
       self.file_name = unicodedata.normalize('NFC', self.file_name) # Normalize unicode
     self.destination = kwargs.get('dest', None)  # GET param avoids infinite looping
-    self.file_name = kwargs.get('qqfilename')
     self._fs = get_client(fs='s3a', user=self._request.user.username)
     self.bucket_name, self.key_name = parse_uri(self.destination)[:2]
     # Verify that the path exists
     self._fs._stats(self.destination)
     self._bucket = self._fs._get_bucket(self.bucket_name)
     self.filepath = self._fs.join(self.key_name, self.file_name)
+    if kwargs.get('chunk_size', None) != None:
```
Comment (Harshg999) [inline desktop/libs/aws/src/aws/s3/upload.py:73]: Using `if kwargs.get('chunk_size'):` should simplify the check?

File: desktop/libs/azure/src/azure/abfs/upload.py [hunk 8]
```diff
@@ -45,35 +46,39 @@
...
     self._request = request
-    self._part_size = DEFAULT_WRITE_SIZE
-
     self.qquuid = kwargs.get('qquuid')
     self.qqtotalparts = kwargs.get('qqtotalparts')
     self.totalfilesize = kwargs.get('qqtotalfilesize')
     self.file_name = kwargs.get('qqfilename')
     if self.file_name:
       self.file_name = unicodedata.normalize('NFC', self.file_name) # Normalize unicode
+    self.destination = kwargs.get('dest', None)  # GET param avoids infinite looping
```
Comment (Harshg999) [inline desktop/libs/azure/src/azure/abfs/upload.py:57]: Only `kwargs.get('dest')` should be good?

File: desktop/libs/azure/src/azure/abfs/upload.py [hunk 9]
```diff
@@ -45,35 +46,39 @@
...
+    self.target_path = None
+
+    if kwargs.get('chunk_size', None) != None:
```
Comment (Harshg999) [inline desktop/libs/azure/src/azure/abfs/upload.py:60]: Using `if kwargs.get('chunk_size'):` should simplify the check?

File: desktop/libs/hadoop/src/hadoop/fs/upload.py [hunk 10]
```diff
@@ -94,9 +91,14 @@ def __init__(self, request, *args, **kwargs):
       self.file_name = unicodedata.normalize('NFC', self.file_name) # Normalize unicode
     self.dest = kwargs.get('dest')
     self.file_name = kwargs.get('qqfilename')
-    self.filepath = request.fs.join(self.dest, self.file_name)
+    if kwargs.get('filepath', None) != None:
```
Comment (Harshg999) [inline desktop/libs/hadoop/src/hadoop/fs/upload.py:94]: Using `if kwargs.get('filepath'):` should simplify the check?

File: desktop/libs/hadoop/src/hadoop/fs/upload.py [hunk 11]
```diff
@@ -94,9 +91,14 @@ def __init__(self, request, *args, **kwargs):
...
+      self.filepath = kwargs.get('filepath')
+    else:
+      self.filepath = request.fs.join(self.dest, self.file_name)
+      kwargs['filepath'] = self.filepath
     self._file = None
-    self.chunk_size = 0
+    if kwargs.get('chunk_size', None) != None:
```
Comment (Harshg999) [inline desktop/libs/hadoop/src/hadoop/fs/upload.py:100]: Using `if kwargs.get('chunk_size'):` should simplify the check?

File: desktop/libs/notebook/src/notebook/models.py [hunk 12]
```diff
@@ -570,15 +570,15 @@ def __init__(self, request, snippet):
...
-        LOG.debug('Skipping Task Server call %s' % name)
+    # if TASK_SERVER.ENABLED.get():
+    #   from notebook import tasks as ntasks
+    #   if hasattr(ntasks, name):
+    #     attr = getattr(ntasks, name)
+    #     def _method(*args, **kwargs):
+    #       return attr(*args, **dict(kwargs, postdict=self.request.POST, user_id=self.request.user.id))
+    #     return _method
+    #   else:
+    #     LOG.debug('Skipping Task Server call %s' % name)
```
Comment (Harshg999) [inline desktop/libs/notebook/src/notebook/models.py:581]: Nit: To remove?

**Review by Harshg999 — CHANGES_REQUESTED** [review]
Added few review comments, also possible to add few unit tests?

**Comment by ranade1** [comment]
> Added few review comments, also possible to add few unit tests?

added a unit test.

**Review by ranade1 — COMMENTED** [review]
added a test case.

**Review by Harshg999 — APPROVED** [review]
LGTM!

**Pull request merged**